docs/examples/kafka-enrichment.md

```diff
@@ -0,0 +1,215 @@
+
+_A Quick Aside on Recovery: With Bytewax you can persist state in more durable formats. This is so that in the case that the dataflow fails, you can recover state and the dataflow will not be required to recompute from the beginning. This is oftentimes referred to as checkpointing for other processing frameworks. With the `KafkaInputConfig` configuring recovery will handle the offset and consumer groups for you. This makes it easy to get started working with data in Kafka._
+
+**Data Transformation**
+
+[Operators](https://docs.bytewax.io/apidocs/bytewax.dataflow) are Dataflow class methods that define how data will flow through the dataflow. Whether it will be filtered, modified, aggregated or accumulated. In this example we are modifying our data in-flight and will use the `map` operator. The `map` operator will receive a Python function as an argument and this will contain the code to modify the data payload.
+
+```python doctest:SKIP
+import json
+import requests
+
+def get_location(data):
+    key, value = data
+    ip_address = value.decode('ascii')
+    response = requests.get(f'https://ipapi.co/{ip_address}/json/')
+    response_json = response.json()
+    location_data = {
+        "ip": ip_address,
+        "city": response_json.get("city"),
+        "region": response_json.get("region"),
+        "country_name": response_json.get("country_name")
+    }
+    return key, json.loads(location_data)
+
+# corresponding flow module for the Map operator
+flow.map(get_location)
+```
+
+In the code above, we are making an http request to an external service, this is only for demonstration purposes. You should use something that is lower latency so you do not risk having the http request as a bottleneck or having network errors.
+
+**Kafka Output**
+
+To capture data that is transformed in a dataflow, the capture method is used.Similarly to the input method, it takes a configuration as the argument. Bytewax has built-in output configurations and [`KafkaOutputConfig`](https://docs.bytewax.io/apidocs/bytewax.outputs#bytewax.outputs.KafkaOutputConfig) is one of those. We are going to use it in this example to write out the enriched data to a new topic.
+
+```python doctest:SKIP
+flow.capture(
+    KafkaOutputConfig(
+        brokers=["localhost:9092"],
+        topic="ip_address_by_location"
+    )
+)
+```
+
+### Kicking off execution
+
+With the above dataflow written the final step is to specify the execution method. Whether it will run as a single threaded process on our local machine or be capable of scaling across a kubernetes cluster. The methods used to define the execution are part of the execution module and more detail can be found in the long format documentation as well as in the API documentation.
```

Comment:
```suggestion
With the above dataflow written, the final step is to specify the execution method. Whether it will run as a single threaded process on our local machine or be capable of scaling across a Kubernetes cluster. The methods used to define the execution are part of the execution module and more detail can be found in the long format documentation as well as in the API documentation.
```